### PR TITLE
[CM H1] Update Buildserver Documentation

### PR DESCRIPTION
**Pull request by robaerd**
## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: doc/BUILDSERVER.md [hunk 1]
```diff
@@ -39,18 +39,50 @@ Summarized libelektra's job purpose is to combine the where
...
 
+#### Jenkinsfile.monthly
+
+- [Jenkinsfile.monthly](/scripts/jenkins/Jenkinsfile.monthly) contains monthly triggering of libelektra job on master.
+- [Buildjob: libelektra-monthly](https://build.libelektra.org/job/libelektra-monthly/)
+- [Jenkinsfile.monthly](https://master.libelektra.org/scripts/jenkins/Jenkinsfile.monthly)
+
+#### Jenkinsfile.release
+
+- Triggered manually with optional arguments and is for automated releasing a new version of Elektra.
```
Comment (kodebach) [inline doc/BUILDSERVER.md:81]:
```suggestion
- Triggered manually with optional arguments. It is used for automated releasing a new version of Elektra.
```

File: doc/BUILDSERVER.md [hunk 2]
```diff
@@ -161,10 +195,35 @@ your needs.
 
 For runs of the build job that are run in the master branch we also execute
 deployment steps after all tests pass.
-We use it to build Debian packages and move it into the repository on the a7
-node.
+We use it to build Debian and Fedora packages and move them into the repositories
```
Comment (kodebach) [inline doc/BUILDSERVER.md:198]:
```suggestion
We use these steps to build Debian and Fedora packages and move them into the repositories
```

File: doc/BUILDSERVER.md [hunk 3]
```diff
@@ -161,10 +195,35 @@ your needs.
...
+The `buildRelease` function runs various test suites and collects and archives debug
+information. It also generates source-packages and packages for Debian based
+distributions and Fedora.
+
+The generated packages are additionally installed and tested on a clean
+Docker image.
+
+### Manual approval
+
+In order to verify the correct behavior and results of the previous steps and its
```
Comment (kodebach) [inline doc/BUILDSERVER.md:216]:
```suggestion
In order to verify the correct behavior and results of the previous steps and their
```

File: doc/BUILDSERVER.md [hunk 4]
```diff
@@ -161,10 +195,35 @@ your needs.
...
+generated artifacts, we use the [Input Step](https://www.jenkins.io/doc/pipeline/steps/pipeline-input-step/)
+to pause the pipeline.
+After manually verifying the correctness, the pipeline run can be resumed.
+
+### Publishing
+
+The previously generated packages are published to its repositories, as described
```
Comment (kodebach) [inline doc/BUILDSERVER.md:223]:
```suggestion
The previously generated packages are published to the repositories, as described
```

What does "its" refer to in "its repositories"?

File: doc/BUILDSERVER.md [hunk 5]
```diff
@@ -161,10 +195,35 @@ your needs.
...
+above in _Jenkinsfile (Main CI Pipeline)_.
+The API documentation and source packages get published to its git repositories.
```
Comment (kodebach) [inline doc/BUILDSERVER.md:225]:
```suggestion
The API documentation and source packages get published to the relevant git repositories.
```

Again what is this "it" you refer to?

**Review by kodebach — COMMENTED** [review]
LGTM. Finally, I understand where those mysterious Docker functions in our Jenkinsfile come from :)
I just suggested a few English language improvements to make things easier to understand.

PS. not sure what is wrong with Travis, but I think we can ignore it for this PR

**Comment by mpranj** [comment]
Looks great, thank you so much!

(ping @markus2330: this is a [CM H1] related merge)

File: doc/BUILDSERVER.md [hunk 6]
```diff
@@ -193,6 +248,124 @@ verified or added to build Elektra correctly:
...
+this `libelektra` repository.
+
+Most of the default settings can be used.
+
+- If a private repository were to be used with a GitHub type job,
+  a credentials pair of `elektrabot` must be selected to access this repository.
+- For Build Configuration you want to specify `by Jenkinsfile` and add the
+  script path to the Jenkinsfile.
+
+We first start by including the shared library as described above in _Jenkins Shared Library_.
```
Comment (markus2330) [inline doc/BUILDSERVER.md:274]: Maybe move "Jenkins Shared Library" down here, it does not seem to be needed before?

File: doc/news/_preparation_next_release.md [hunk 7]
```diff
@@ -203,7 +203,7 @@ _(@tucek)_
 
 - <<TODO>>
 - <<TODO>>
-- <<TODO>>
+- Update buildserver documentation _(Robert Sowula)_
```
Comment (markus2330) [inline doc/news/_preparation_next_release.md:206]: Could be more detailed. Please also link to the document.

**Review by markus2330 — COMMENTED** [review]
Thank you, very nicely done. Very good to have this docu. A few small notes for a follow-up PR.

**Pull request merged**